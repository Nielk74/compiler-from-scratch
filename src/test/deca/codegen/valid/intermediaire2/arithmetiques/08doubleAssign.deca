// Description:
//    Programme testant l'affectation avec une autre affectation
//
// Resultats:
//    5
//    7
//    4.00000e+00
//    7.00000e+00
//
// Historique:
//    cree le 10/06/2022

{
    int a = 8;
    int b = (a = 5);
    float c = (a = 4);
    float d;
    int e = 7;

    println(b);
    b = (a = e);
    println(b);

    println(c);
    c = (d = a);
    println(c);
}