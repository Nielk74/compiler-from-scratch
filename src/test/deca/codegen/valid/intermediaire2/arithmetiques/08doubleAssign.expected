5
7
4.00000e+00
7.00000e+00
